packages/core/src/tools/utils.ts

```diff
@@ -388,6 +388,148 @@ export function runOnReadyState(expectedReadyState: 'complete' | 'interactive',
   }
 }
 
+type Merged<TDestination, TSource> =
```

Comment:
Yes, it should - but it will not be easy to do. I did a quick pass and it will also introduce some type errors that will require further changes in type annotations. Could we defer this change to a separate PR?

Reply:
Sure! That's pretty neat already :)